### ISSUE TITLE
Copr/Koji tmt plugin

### ISSUE THREAD

**Issue by LecrisUT** [teemtee/tmt#2859]
## Goal

Have a plugin for creating `copr`/`koji` reverse dependency testing

## Design

- [ ] `plan.discover`: Create tests that would run `copr`/`koji` builds
  - `packages`: Packages for which to run `fedrq wr ${packages}`/`fedrq wr --src ${packages}` (it seems to give union instead of intersect)
  - `spec_file`: Alternative to the above that generates all `packages` from the spec file
  - `...`: Other args to be passed to generate the test framework with
- [ ] `test.framework`: Main test interface running `copr`/`koji` build
  - `user/group`: copr user or group to which to submit the build
  - `project`: copr project where to build in
  - `from_dist_git`: Create copr build using the distgit repo's spec file
  - For koji, the builds would just be `--scratch`
- [ ] Authentication: Could be handled manually by a `prepare` step  or by the infrastructure `testing-farm`/`github` secrets
- [ ] How to include the current build?
  - Manual via `copr`, assume this is already setup in the `user/group/project` variables
  - `koji_build`/`copr_project`: variables pointing to where the local build was created
  - `testing-farm` and `packit` provide the information of where the build was made, but it would hard-code the integration there

## Future improvements

- Integration with upstream/downstream: E.g. building in side-tag, specifying dependent git repo/branch
- Chaining `testing-farm` executions such that the integration tests are run for upstream
- Packit interface to use in `test.framework`
- Create copr build from packit.yaml if downstream or upstream contains the configuration
- Equivalent integration for local build with `mock`

**Comment by psss** [comment]
@LecrisUT, is this one related to:

* #3990 

Or rather a completely different thing?

**Comment by LecrisUT** [comment]
> [@LecrisUT](https://github.com/LecrisUT), is this one related to:

Not quite. It would be the inverse of it, specifically running koji/copr builds as a test framework, not making the artifacts available for tests, would leave that as a responsibility for the user for now.

**Comment by LecrisUT** [comment]
This is still kinda relevant because there are some tests that do reverse-dependency builds. Although it is more complex and probably leaving the user to use cli commands is better for this one because there are too many complications with side-tags and such.
